Add javascript documentation for Crons upsert (#8723)
getsentry/sentry-docs: src/includes/javascript-crons-upsert.mdx

You can create and update your Monitors programmatically with code rather than [creating and configuring them in Sentry.io](https://sentry.io/crons/create/).

> Using `Sentry.withMonitor()`, you can pass a third parameter with your monitor configuration:

Reviewer: Is this creating the monitor? Can we reword this to be more action-oriented?

```suggestion
To create/update a monitor, use `Sentry.withMonitor()` and pass in your monitor configuration as a third parameter:
```

```javascript
Sentry.withMonitor(
  "<monitor-slug>",
  () => {
    // Execute your scheduled task here...
  },
  {
    schedule: {
      type: "crontab",
      value: "* * * * *",
    },
    checkinMargin: 2, // In minutes. Optional.
    maxRuntime: 10, // In minutes. Optional.
    timezone: "America/Los_Angeles", // Optional.

>   }

Reviewer: Would it make sense to define monitorConfig separately and then pass it in, as done in the next code sample? I think this would make it clearer what this sample is showing.

Author: Done

);
```

> Using `Sentry.captureCheckIn()`, you can pass a second parameter with your monitor configuration:

Reviewer: ```suggestion
To configure the monitor's check-ins, use `Sentry.captureCheckIn()` and pass in your monitor configuration as a second parameter:
```

};

// 🟡 Notify Sentry your job is running:
const checkInId = Sentry.captureCheckIn(
  {
    monitorSlug: "<monitor-slug>",
    status: "in_progress",
  },
  monitorConfig
);

// Execute your scheduled task here...

// 🟢 Notify Sentry your job has completed successfully:
Sentry.captureCheckIn(
  {
    // Make sure this variable is named `checkInId`
    checkInId,
    monitorSlug: "<monitor-slug>",
    status: "ok",
  },
  monitorConfig
);
```

```typescript
import { MonitorConfig } from "@sentry/types";

const monitorConfig: MonitorConfig = {
  schedule: {
    type: "crontab",
    value: "* * * * *",
  },
  checkinMargin: 2, // In minutes. Optional.
  maxRuntime: 10, // In minutes. Optional.
  timezone: "America/Los_Angeles", // Optional.
};

// 🟡 Notify Sentry your job is running:
const checkInId = Sentry.captureCheckIn(
  {
    monitorSlug: "<monitor-slug>",
    status: "in_progress",
  },
  monitorConfig
);

// Execute your scheduled task here...

// 🟢 Notify Sentry your job has completed successfully:
Sentry.captureCheckIn(
  {
    // Make sure this variable is named `checkInId`
    checkInId,
    monitorSlug: "<monitor-slug>",
    status: "ok",
  },
  monitorConfig
);
```

> Available monitor config properties:

Reviewer: ```suggestion
### Monitor Configuration Properties

 The following are available monitor configuration properties:
```

`schedule`:

: The job's schedule :

> The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:
> 
> ```json
> {"type": "crontab", "value": "0 * * * *"}
> {"type": "interval", "value": "2", "unit": "hour"}
> ```

Reviewer: ```suggestion
  The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:
  
  ```json
  {"type": "crontab", "value": "0 * * * *"}
  {"type": "interval", "value": "2", "unit": "hour"}
  ```
```

Small nit, but can we tab these in a bit so the info organization is visually clearer?

    monitorSlug: "<monitor-slug>",
    status: "ok",
  },
  monitorConfig
);
```

```typescript
import { MonitorConfig } from "@sentry/types";

const monitorConfig: MonitorConfig = {
  schedule: {
    type: "crontab",
    value: "* * * * *",
  },
  checkinMargin: 2, // In minutes. Optional.
  maxRuntime: 10, // In minutes. Optional.
  timezone: "America/Los_Angeles", // Optional.
};

// 🟡 Notify Sentry your job is running:
const checkInId = Sentry.captureCheckIn(
  {
    monitorSlug: "<monitor-slug>",
    status: "in_progress",
  },
  monitorConfig
);

// Execute your scheduled task here...

// 🟢 Notify Sentry your job has completed successfully:
Sentry.captureCheckIn(
  {
    // Make sure this variable is named `checkInId`
    checkInId,
    monitorSlug: "<monitor-slug>",
    status: "ok",
  },
  monitorConfig
);
```

### Monitor Configuration Properties

The following are available monitor configuration properties:

`schedule`:

: The job's schedule :

The schedule representation for your monitor, either `crontab` or `interval`. The structure will vary depending on the type:

```json
{"type": "crontab", "value": "0 * * * *"}
{"type": "interval", "value": "2", "unit": "hour"}

> 

Reviewer: ```suggestion
```

```
Looks like this code block ending got moved

`checkinMargin`:

: The amount of time (in minutes) Sentry should wait for your check-in before it's considered missed ("grace period"). Optional.

<Note>

We recommend that your check-in margin be less than or equal to your interval.

</Note>

`maxRuntime`:

: The amount of time (in minutes) your job is allowed to run before it's considered failed. Optional.

> `timezone`

Reviewer: ```suggestion
`timezone`:
```

: The `tz` where your job is running. This is usually your server's timezone, (such as `America/Los_Angeles`). See [list of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones). Optional.

> ```

Reviewer: ```suggestion
```
